src/core/deadbeef.d

```diff
@@ -0,0 +1,141 @@
+ * of the function.
+ * Useful for:
+ *
+ * 1. testing the code generator
+ * 2. avoiding leaking information that should be hidden
+ * from the caller
+ *
+ * Inserting a call to this function at any point in the code
+ * should not produce any failures.
+ */
+
+nothrow
+@safe
+@nogc
+pure
+void deadbeef()
+{
+    /* Rely on the inline assembler to know which registers
+     * are to be preserved by the ABI.
+     */
+    version (D_InlineAsm_X86)
+    {
+        asm pure nothrow @safe @nogc
+        {
+            mov EAX,0xDEADBEEF  ;
+            mov EBX,EAX         ;
+            mov ECX,EAX         ;
+            mov EDX,EAX         ;
+            //mov EBP,EAX       ;
+            mov ESI,EAX         ;
+            mov EDI,EAX         ;
+
+            push EAX            ;
+            movd XMM0,[ESP]     ;
+            pshufd XMM0,XMM0,0  ;
+            movdqa XMM1,XMM0    ;
+            movdqa XMM2,XMM0    ;
+            movdqa XMM3,XMM0    ;
+            movdqa XMM4,XMM0    ;
+            movdqa XMM5,XMM0    ;
+            movdqa XMM6,XMM0    ;
+            movdqa XMM7,XMM0    ;
+            pop EAX             ;
+
+            fldlg2              ;
+            fldlg2              ;
+            fldlg2              ;
+            fldlg2              ;
+            fldlg2              ;
+            fldlg2              ;
+            fldlg2              ;
+            fldlg2              ;
+            fstp ST(0)          ;
+            fstp ST(0)          ;
+            fstp ST(0)          ;
+            fstp ST(0)          ;
+            fstp ST(0)          ;
+            fstp ST(0)          ;
+            fstp ST(0)          ;
+            fstp ST(0)          ;
+        }
+    }
+    else
+    version (D_InlineAsm_X86_64)
+    {
+        asm pure nothrow @safe @nogc
+        {
+            mov RAX,0xDEADBEEFDEADBEEF  ;
+            mov RBX,RAX         ;
+            mov RCX,RAX         ;
+            mov RDX,RAX         ;
+            mov RBP,RAX         ;
+            mov RSI,RAX         ;
+            mov RDI,RAX         ;
+            mov R8,RAX          ;
+            mov R9,RAX          ;
+            mov R10,RAX         ;
+            mov R11,RAX         ;
+            mov R12,RAX         ;
+            mov R13,RAX         ;
+            mov R14,RAX         ;
+            mov R15,RAX         ;
+
+            push RAX            ;
+            movd XMM0,[RSP]     ;
+            pshufd XMM0,XMM0,0  ;
+            movdqa XMM1,XMM0    ;
+            movdqa XMM2,XMM0    ;
+            movdqa XMM3,XMM0    ;
+            movdqa XMM4,XMM0    ;
+            movdqa XMM5,XMM0    ;
+            movdqa XMM6,XMM0    ;
+            movdqa XMM7,XMM0    ;
+            movdqa XMM8,XMM0    ;
+            movdqa XMM9,XMM0    ;
+            movdqa XMM10,XMM0   ;
+            movdqa XMM11,XMM0   ;
+            movdqa XMM12,XMM0   ;
+            movdqa XMM13,XMM0   ;
+            movdqa XMM14,XMM0   ;
+            movdqa XMM15,XMM0   ;
```

Comment:
What about YMM registers for AVX?

Reply:
I thought for the first version I'd stick with the baseline.